Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -0,0 +1,280 @@
+                ofs << m_sep;
+                ofs << "[" << c++ << "]probe_Ey_lev" + std::to_string(lev) + " (V/m)";
+                ofs << m_sep;
+                ofs << "[" << c++ << "]probe_Ez_lev" + std::to_string(lev) + " (V/m)";
+                ofs << m_sep;
+                ofs << "[" << c++ << "]probe_|E|_lev" + std::to_string(lev) + " (V/m)";
+                ofs << m_sep;
+                ofs << "[" << c++ << "]probe_Bx_lev" + std::to_string(lev) + " (T)";
+                ofs << m_sep;
+                ofs << "[" << c++ << "]probe_By_lev" + std::to_string(lev) + " (T)";
+                ofs << m_sep;
+                ofs << "[" << c++ << "]probe_Bz_lev" + std::to_string(lev) + " (T)";
+                ofs << m_sep;
+                ofs << "[" << c++ << "]probe_|B|_lev" + std::to_string(lev) + " (T)";
+            }
+            ofs << std::endl;
+            // close file
+            ofs.close();
+        }
+    }
+}
+// end constructor
+
+// function that computes maximum field values
+void FieldProbe::ComputeDiags (int step)
+{
+    // Judge if the diags should be done
+    if (!m_intervals.contains(step+1)) { return; }
+
+    // get a reference to WarpX instance
+    auto & warpx = WarpX::GetInstance();
+
+    // get number of level
+    const auto nLevel = warpx.finestLevel() + 1;
+
+    // loop over refinement levels
+    for (int lev = 0; lev < nLevel; ++lev)
+    {
+
+        const amrex::Geometry& gm = warpx.Geom(lev);
+        const auto prob_lo = gm.ProbLo();
+        const auto prob_hi = gm.ProbHi();
+
+        const bool probe_in_domain = x_probe >= prob_lo[0] and y_probe >= prob_lo[1] and
+                                     z_probe >= prob_lo[2] and x_probe <= prob_hi[0] and
+                                     y_probe <= prob_hi[1] and z_probe <= prob_hi[2];
+
+        amrex::Real fp_Ex, fp_Ey, fp_Ez, fp_E, fp_Bx, fp_By, fp_Bz, fp_B;
+
+        if( probe_in_domain ) {
+            const auto cell_size = gm.CellSizeArray();
+
+            const int i_probe = amrex::Math::floor((x_probe - prob_lo[0]) / cell_size[0]);
+            const int j_probe = amrex::Math::floor((y_probe - prob_lo[1]) / cell_size[1]);
+            const int k_probe = amrex::Math::floor((z_probe - prob_lo[2]) / cell_size[2]);
+
+            // get MultiFab data at lev
+            const amrex::MultiFab &Ex = warpx.getEfield(lev, 0);
+            const amrex::MultiFab &Ey = warpx.getEfield(lev, 1);
+            const amrex::MultiFab &Ez = warpx.getEfield(lev, 2);
+            const amrex::MultiFab &Bx = warpx.getBfield(lev, 0);
+            const amrex::MultiFab &By = warpx.getBfield(lev, 1);
+            const amrex::MultiFab &Bz = warpx.getBfield(lev, 2);
+
+            // Prepare interpolation of field components to cell center
+            // The arrays below store the index type (staggering) of each MultiFab, with the third
+            // component set to zero in the two-dimensional case.
```

Comment:
```suggestion
            // The arrays below store the index type (staggering) of each MultiFab.
```

Since you're using an `IndexType` here, there should be no third component in 2D.